language/compiler/ir_to_bytecode/syntax/src/ast.rs

```diff
@@ -523,6 +526,23 @@ impl Script {
             FunctionBody::Native => panic!("main() can't be native"),
         }
     }
+
+    /// Return a vector of `AccessPath` where the on chain dependent modules are stored.
+    pub fn get_on_chain_deps(&self) -> Vec<AccessPath> {
+        let mut deps = vec![];
+        for dep in self.imports.iter() {
+            match &dep.ident {
+                ModuleIdent::Qualified(id) => {
+                    deps.push(AccessPath::code_access_path(&ModuleId::new(
+                        id.address,
+                        id.name.name(),
+                    )));
+                }
+                _ => (),
```

Comment:
Nit: explicit pattern matching is safe. It lets the compiler tell us when we add variants that need to be handled here in the future.